com.kaii.hgui.json

```diff
@@ -0,0 +1,65 @@
+{
+    "app-id" : "com.kaii.hgui",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "master",
+    "sdk" : "org.gnome.Sdk",
+    "sdk-extensions" : [
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "hgui-rust",
+    "finish-args" : [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland"
+
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+        ],
+        "env" : {
+            "RUST_BACKTRACE" : "1",
+            "RUST_LOG" : "hgui-rust=debug",
+            "CARGO_HOME" : "/run/build/hgui-rust/cargo"
+        }
+    },
+    "cleanup" : [
+        "/include",
+        "/lib/pkgconfig",
+        "/man",
+        "/share/doc",
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+          "name": "blueprint-compiler",
+          "buildsystem": "meson",
+          "cleanup": ["*"],
+          "sources": [
+            {
+              "type": "git",
+              "url": "https://gitlab.gnome.org/jwestman/blueprint-compiler.git",
+              "tag": "v0.10.0"
+            }
+          ]
+        },
+        {
+            "name" : "hgui-rust",
+            "builddir" : true,
+            "buildsystem" : "meson",
```

Comment:
This won't work, your meson https://github.com/kaii-lb/hgui-rust/blob/master/src/meson.build isn't running something like 

```
 cargo --offline fetch --manifest-path root/Cargo.toml \
            && cargo --offline build --release 
```

It ignores the crates in the cargo manifest:

```
[0/5] Generating src/cargo-build with a custom command
    Updating crates.io index
warning: spurious network error (3 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
warning: spurious network error (2 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
warning: spurious network error (1 tries remaining): [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
error: failed to get `libadwaita` as a dependency of package `hgui-rust v1.0.2 (/run/build/hgui-rust)`

Caused by:
  failed to query replaced source registry `crates-io`

Caused by:
  download of config.json failed

Caused by:
  failed to download from `https://index.crates.io/config.json`

Caused by:
  [6] Couldn't resolve host name (Could not resolve host: index.crates.io)
```

Either change it to simple and build manually or add a flatpak target upstream.

This'll build the crates

```
            "buildsystem": "simple",
            "build-commands": [
                "cargo --offline fetch --manifest-path Cargo.toml --verbose",
                "cargo --offline build --release --verbose"
            ]
```

Reply:
if i use the simple build system, i would have to install every icon/desktop/appdata file by hand, that might not be the best choice. i'll try to modify the meson build target to do this instead

Reply:
with the `generated-sources.json` it's supposed to work.

BUT if you use crates from git you might end up in trouble.

Reply:
closer but still, im getting error: `no matching package named 'arboard' found`